Git commit unit test flakes

I've encountered some local flakes with `TestGitCommit_GenerateFullyQualifiedImageName`

These fail for me every time locally on both OS X and Linux. Interestingly, the short hash that the test actually computes is the same on both platforms:

```
--- FAIL: TestGitCommit_GenerateFullyQualifiedImageName (0.02s)
    --- FAIL: TestGitCommit_GenerateFullyQualifiedImageName/success (0.01s)
    	util.go:51: string differ.
    		Expected
    		test:41cf71e
    		Actual
    		test:0e78a85
```

@dgageot both of those fail actually, as do `TestGitCommit_GenerateFullyQualifiedImageName/untracked` and `TestGenerateFullyQualifiedImageNameFromSubDirectory`:
```
--- FAIL: TestGitCommit_GenerateFullyQualifiedImageName (0.05s)
    --- FAIL: TestGitCommit_GenerateFullyQualifiedImageName/success (0.03s)
    	util.go:51: string differ.
    		Expected
    		test:41cf71e
    		Actual
    		test:0e78a85
    --- FAIL: TestGitCommit_GenerateFullyQualifiedImageName/dirty (0.02s)
    	util.go:51: string differ.
    		Expected
    		test:41cf71e-dirty-17c3e6fb2811b7af
    		Actual
    		test:0e78a85-dirty-17c3e6fb2811b7af
    --- FAIL: TestGitCommit_GenerateFullyQualifiedImageName/untracked (0.01s)
    	util.go:51: string differ.
    		Expected
    		test:41cf71e-dirty-d7bc32e5f6760a99
    		Actual
    		test:0e78a85-dirty-d7bc32e5f6760a99
    --- PASS: TestGitCommit_GenerateFullyQualifiedImageName/failure (0.00s)
=== RUN   TestGenerateFullyQualifiedImageNameFromSubDirectory
--- FAIL: TestGenerateFullyQualifiedImageNameFromSubDirectory (0.01s)
	git_commit_test.go:122: Invalid names found: test:c75d899 and test:c75d899
```

This is at `HEAD`

@dgageot ran `make test` on your branch on my mac:

```
--- FAIL: TestGitCommit_GenerateFullyQualifiedImageName (0.23s)
	git_commit_test.go:176: 97d24779af95532b316ebd753575c5f775498eb3 95a5f48febce6b1dc60518bbd568f210b267420f
    --- FAIL: TestGitCommit_GenerateFullyQualifiedImageName/success (0.04s)
    	util.go:51: string differ.
    		Expected
    		test:97d2477
    		Actual
    		test:95a5f48
	git_commit_test.go:176: 97d24779af95532b316ebd753575c5f775498eb3 95a5f48febce6b1dc60518bbd568f210b267420f
    --- FAIL: TestGitCommit_GenerateFullyQualifiedImageName/dirty (0.08s)
    	util.go:51: string differ.
    		Expected
    		test:97d2477-dirty-17c3e6fb2811b7af
    		Actual
    		test:95a5f48-dirty-17c3e6fb2811b7af
	git_commit_test.go:176: 97d24779af95532b316ebd753575c5f775498eb3 95a5f48febce6b1dc60518bbd568f210b267420f
    --- FAIL: TestGitCommit_GenerateFullyQualifiedImageName/untracked (0.11s)
    	util.go:51: string differ.
    		Expected
    		test:97d2477-dirty-d7bc32e5f6760a99
    		Actual
    		test:95a5f48-dirty-d7bc32e5f6760a99
    --- PASS: TestGitCommit_GenerateFullyQualifiedImageName/failure (0.00s)
=== RUN   TestGenerateFullyQualifiedImageNameFromSubDirectory
--- FAIL: TestGenerateFullyQualifiedImageNameFromSubDirectory (0.05s)
	git_commit_test.go:176: 6e0beaba915387e484980b9c3bd77e497947c36e c75d89960788c31743ec39b0a315ab3f601ef964
	git_commit_test.go:122: Invalid names found: test:c75d899 and test:c75d899
```
also verified I get the same error after deleting my `$HOME/.gitconfig`.

--- a/pkg/skaffold/build/tag/git_commit_test.go
+++ b/pkg/skaffold/build/tag/git_commit_test.go
@@ -42,7 +42,7 @@ func TestGitCommit_GenerateFullyQualifiedImageName(t *testing.T) {
 				ImageName: "test",
 				Digest:    "sha256:12345abcde",
 			},
-			expectedName: "test:41cf71e",
+			expectedName: "test:eefe1b9",
 			createGitRepo: func(dir string) {
 				gitInit(t, dir).
 					write(t, "source.go", []byte("code")).
@@ -55,7 +55,7 @@ func TestGitCommit_GenerateFullyQualifiedImageName(t *testing.T) {
 			opts: &TagOptions{
 				ImageName: "test",
 			},
-			expectedName: "test:41cf71e-dirty-17c3e6fb2811b7af",
+			expectedName: "test:eefe1b9-dirty-17c3e6fb2811b7af",
 			createGitRepo: func(dir string) {
 				gitInit(t, dir).
 					write(t, "source.go", []byte("code")).
@@ -69,7 +69,7 @@ func TestGitCommit_GenerateFullyQualifiedImageName(t *testing.T) {
 			opts: &TagOptions{
 				ImageName: "test",
 			},
-			expectedName: "test:41cf71e-dirty-d7bc32e5f6760a99",
+			expectedName: "test:eefe1b9-dirty-d7bc32e5f6760a99",
 			createGitRepo: func(dir string) {
 				gitInit(t, dir).
 					write(t, "source.go", []byte("code")).
@@ -118,7 +118,7 @@ func TestGenerateFullyQualifiedImageNameFromSubDirectory(t *testing.T) {
 	name2, err := c.GenerateFullyQualifiedImageName(subDir, opts)
 	failNowIfError(t, err)
 
-	if name1 != name2 || name1 != "test:6e0beab" {
+	if name1 != name2 || name1 != "test:a7b32a6" {
 		t.Errorf("Invalid names found: %s and %s", name1, name2)
 	}
 }
@@ -148,7 +148,7 @@ func (g *gitRepo) mkdir(folder string) *gitRepo {
 }
 
 func (g *gitRepo) write(t *testing.T, file string, content []byte) *gitRepo {
-	err := ioutil.WriteFile(filepath.Join(g.dir, file), content, 0644)
+	err := ioutil.WriteFile(filepath.Join(g.dir, file), content, os.ModePerm)
 	failNowIfError(t, err)
 	return g
 }
@@ -160,7 +160,7 @@ func (g *gitRepo) add(t *testing.T, file string) *gitRepo {
 }
 
 func (g *gitRepo) commit(t *testing.T, msg string) *gitRepo {
-	now, err := time.Parse("Jan 2, 2006 at 3:04pm (MST)", "Feb 3, 2013 at 7:54pm (PST)")
+	now, err := time.Parse("Jan 2, 2006 at 15:04:05 -0700 MST", "Feb 3, 2013 at 19:54:00 -0700 MST")
 	failNowIfError(t, err)
 
 	_, err = g.workTree.Commit(msg, &git.CommitOptions{

